src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStream.cs

```diff
@@ -215,13 +183,14 @@ public virtual void QueueCancellation()
         public override async ValueTask DisposeAsync()
         {
             await _writeStream.DisposeAsync().ConfigureAwait(false);
-            // TODO: For Read Stream as well.
+            await _readStream.DisposeAsync().ConfigureAwait(false);
         }
 
         /// <inheritdoc />
         protected override void Dispose(bool disposing)
```

Comment:
I'm not an expert on this, but from what I've seen there's a pattern that disposing should be checked for true before calling dispose, and the underlying shouldn't be disposed if the object was already disposed. May be moot if the underlying will check this, but I want to make sure we follow the best practices for disposal.